storage/client/client.go

```diff
@@ -261,6 +261,7 @@ func (c *StorageClient) Transactions(ctx context.Context, p apiTypes.GetConsensu
 	}
 	for res.rows.Next() {
 		var t Transaction
+		var code uint64
 		var module string
 		var message string
```

Comment:
These fields are [optional in the db](https://github.com/oasisprotocol/oasis-indexer/blob/5d6573a83d14960994870774829e6749e7ea727e/storage/migrations/01_consensus.up.sql#L53-L55)

urg... though I see we [always set them](https://github.com/oasisprotocol/oasis-indexer/blob/5d6573a83d14960994870774829e6749e7ea727e/analyzer/consensus/consensus.go#L431-L434) and just use the empty string to denote what should be NULL. Code is always present (and 0 actively denotes success, as opposed to NULL, which would denote "we don't know"), but module and message should ideally be nullable.

Can you please clean that up too? Or we can do it separately if you want to get this PR in. Shouldn't be a lot of changes though:
  - Update table definition: code is NOT NULL
  - Scan module and message into `*string` here
  - Update the code that sets these fields so it maps an empty string to NULL (you pass a `*string` to the query)
  - Mark module and message as not `required` in v1.yaml (that's what got me to start researching this; I think in practice they will always be set for a non-zero `code`, but we don't have guarantees)

Reply:
Good thread pulling. 🧵🔍🐈‍⬛ 

> Update table definition: code is NOT NULL

Agree.

> Scan module and message into *string here

Cool.

> Update the code that sets these fields so it maps an empty string to NULL (you pass a *string to the query)

Can do.

> Mark module and message as not required in v1.yaml (that's what got me to start researching this; I think in practice they will always be set for a non-zero code, but we don't have guarantees)

Ugh. This is truly unfortunate, but fair enough. I am bought in in our `omitempty` scheme.